### ISSUE TITLE
Problem when S is null, occurs when a node is empty

### ISSUE THREAD

**Issue by Zhoot** [AngelQuirogaM/NiceJson#3]
When there's a empty node "{}" in the code, S will be null and this code fails.

(Line 211) in NiceJson.json

```
protected static string UnescapeString(string s)
{
    string result = string.Empty;

    for (int i=0;i<s.Length;i++)
```

**Comment by AngelQuirogaM** [comment]
I have been testing it , but no exceptions has happened at least with this test:

string jsonEmptyCurly = "{\"testing\":{}}";
JsonNode.ParseJsonString(jsonEmptyCurly);

string jsonEmptyCurly = "{}";
JsonNode.ParseJsonString(jsonEmptyCurly);

Please ¿Could you provide an example when it fails?

**Comment by Zhoot** [comment]
![image](https://cloud.githubusercontent.com/assets/4124559/18255771/4bbae682-73ad-11e6-9b0f-091213737295.png)

https://gist.github.com/Zhoot/c1fbb92fa86fd351d9ca233c2ada1e2c#file-save2-json-L22

**Comment by AngelQuirogaM** [comment]
Downloaded, tested and printed out using Unity3D. All seems to be working OK.

By your error number it could be that you have an old version of NiceJson.cs . Could you download it again and/or execute test scene in the test project?

If you are not using Unity3D in order to figure it out what is happening please tell me, What is your development platform?

**Comment by Zhoot** [comment]
Hey
Sorry was a bit busy.
I downloaded the latest version (Was on 1.2 when I found the errors) and I guess you fixed it in 1.3. Either on purpose or not 👍  Nice tool mate. I love it.
